core/store/eth_client.go

```diff
@@ -12,6 +12,23 @@ import (
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
+//go:generate mockgen -package=mocks -destination=../internal/mocks/eth_wrapper_mocks.go github.com/smartcontractkit/chainlink/core/store EthWrapper
+
+// EthWrapper is the interface supplied by EthClient
+type EthWrapper interface {
+	GetNonce(address common.Address) (uint64, error)
+	GetWeiBalance(address common.Address) (*big.Int, error)
+	GetEthBalance(address common.Address) (*assets.Eth, error)
+	GetERC20Balance(address common.Address, contractAddress common.Address) (*big.Int, error)
+	SendRawTx(hex string) (common.Hash, error)
+	GetTxReceipt(hash common.Hash) (*models.TxReceipt, error)
+	GetBlockByNumber(hex string) (models.BlockHeader, error)
+	GetLogs(q ethereum.FilterQuery) ([]models.Log, error)
+	GetChainID() (*big.Int, error)
+	SubscribeToLogs(channel chan<- models.Log, q ethereum.FilterQuery) (models.EthSubscription, error)
+	SubscribeToNewHeads(channel chan<- models.BlockHeader) (models.EthSubscription, error)
+}
```

Comment:
I feel a more descriptive name could be used, though I'm struggling to come up with a suggestion.  Maybe something like 'EthProvider' or 'EthServicer' or 'EthQuerier'?